src/ModelContextProtocol.Core/RequestOptions.cs

```diff
@@ -58,5 +68,33 @@ public JsonObject? Meta
     /// <summary>
     /// The progress token for tracking long-running operations.
     /// </summary>
-    public ProgressToken? ProgressToken { get; set; }
+    public ProgressToken? ProgressToken {
+        get
+        {
+            return _meta?["progressToken"] switch
+            {
+                JsonValue v when v.TryGetValue(out string? s) => new ProgressToken(s),
+                JsonValue v when v.TryGetValue(out long l) => new ProgressToken(l),
+                _ => null
+            };
+        }
+        set
+        {
+            if (value?.Token is {} token)
+            {
+                _meta ??= new JsonObject();
+                _meta["progressToken"] = token switch
+                {
+                    null => _meta.Remove("progressToken"),
```

Comment:
This looks like a bug. _meta.Remove returns a bool, so this is effectively doing:
```csharp
_meta.Remove("progressToken");
_meta["progressToken"] = true;
```
Please ensure there are tests covering all of these cases.